### PR TITLE
Customer edit

### PR DESCRIPTION
**Pull request by Cimanel**
- [x] Edit customer
- [x] Aside with history
image

### CODE REVIEW

File: src/ressources/customers/CustomerEdit.tsx [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+                  color="secondary"
+                />
+              </Box>
+              <Box flex={1} ml={{ xs: 0, sm: "0.5em" }}>
+                <TextInput source="last_name" isRequired fullWidth />
+              </Box>
+            </Box>
+            <TextInput type="email" source="email" isRequired fullWidth />
+            <DateInput source="birthday" />
+            <Box m={1} />
```
Comment (djhi) [inline src/ressources/customers/CustomerEdit.tsx]: You probably don't need this one. All react-admin components should accept a `sx` prop as well

File: src/ressources/customers/CustomerEdit.tsx [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+            </Typography>
+            <TextInput
+              source="address"
+              multiline
+              fullWidth
+              helperText={false}
+            />
+            <Box display={{ xs: "block", sm: "flex" }}>
+              <Box flex={2} mr={{ xs: 0, sm: "0.5em" }}>
+                <TextInput source="city" fullWidth helperText={false} />
```
Comment (djhi) [inline src/ressources/customers/CustomerEdit.tsx]: same

File: src/ressources/customers/CustomerEdit.tsx [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+              </Box>
+              <Box flex={1} mr={{ xs: 0, sm: "0.5em" }}>
+                <TextInput source="stateAbbr" fullWidth helperText={false} />
```
Comment (djhi) [inline src/ressources/customers/CustomerEdit.tsx]: same

File: src/ressources/customers/CustomerEdit.tsx [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+              </Box>
+              <Box flex={2}>
+                <TextInput source="zipcode" fullWidth helperText={false} />
```
Comment (djhi) [inline src/ressources/customers/CustomerEdit.tsx]: same

File: src/ressources/customers/CustomerEdit.tsx [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+              </Box>
+            </Box>
+            <Box m={1} />
```
Comment (djhi) [inline src/ressources/customers/CustomerEdit.tsx]: here too

File: src/ressources/customers/CustomerEdit.tsx [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+            <Typography variant="h6" gutterBottom>
+              Password
+            </Typography>
+            <Box display={{ xs: "block", sm: "flex" }}>
+              <Box flex={1} mr={{ xs: 0, sm: "0.5em" }}>
```
Comment (djhi) [inline src/ressources/customers/CustomerEdit.tsx]: same

File: src/ressources/customers/CustomerEdit.tsx [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+                <PasswordInput source="password" fullWidth />
+              </Box>
+              <Box flex={1} ml={{ xs: 0, sm: "0.5em" }}>
```
Comment (djhi) [inline src/ressources/customers/CustomerEdit.tsx]: same

File: src/ressources/customers/CustomerEdit.tsx [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+                <PasswordInput source="new_password" fullWidth />
+              </Box>
+            </Box>
+            <Box m={1} />
```
Comment (djhi) [inline src/ressources/customers/CustomerEdit.tsx]: same

**Review by djhi — CHANGES_REQUESTED** [review]
Looks good so far!

**Comment by djhi** [comment]
Can you add a screenshot in the PR description?

File: src/ressources/customers/CustomerEdit.tsx [hunk 9]
```diff
@@ -0,0 +1,148 @@
...
+  TextInput,
+  useRecordContext,
+  UseRecordContextParams,
+} from "react-admin";
+import { FieldErrors, FieldValues } from "react-hook-form";
+import { Aside } from "./Aside";
+import { Customer } from "./Customers";
+
+const CustomerTitle = (props: UseRecordContextParams<Customer>) => {
+  const record = useRecordContext<Customer>(props);
```
Comment (slax57) [inline src/ressources/customers/CustomerEdit.tsx]: You should not need to pass down `props` here as well. The `useRecordContext` hook will fetch the context for you. Can you try and confirm your code works without doing this?

File: src/ressources/customers/CustomerEdit.tsx [hunk 10]
```diff
@@ -0,0 +1,148 @@
...
+  const errors: FieldErrors = {};
+  if (!values.first_name) {
+    errors.first_name = "The firstName is required";
+  }
+  if (!values.last_name) {
+    errors.last_name = "The lastName is required";
+  }
+  if (!values.email) {
+    errors.email = "The email is required";
+  }
```
Comment (slax57) [inline src/ressources/customers/CustomerEdit.tsx]: Why do you need this code? Is it a test you made to see if you could override the default behaviour?

File: src/ressources/customers/CustomerEdit.tsx [hunk 11]
```diff
@@ -0,0 +1,148 @@
...
+            </Typography>
+            <Box display={{ xs: "block", sm: "flex" }}>
+              <PasswordInput
+                sx={{ flex: 1, mr: { xs: 0, sm: "0.5em" } }}
+                source="password"
+                fullWidth
+              />
+              <PasswordInput
+                sx={{ flex: 1, ml: { xs: 0, sm: "0.5em" } }}
+                source="new_password"
```
Comment (slax57) [inline src/ressources/customers/CustomerEdit.tsx:116]: Do you check that `new_password` matches `password`?

File: src/ressources/customers/ReviewsNbField.tsx [hunk 12]
```diff
@@ -0,0 +1,31 @@
...
+import CommentIcon from "@mui/icons-material/Comment";
+import {
+  useGetList,
+  useRecordContext,
+  UseRecordContextParams,
+} from "react-admin";
+import { Customer } from "./Customers";
+
+export const ReviewsNbField = (props: UseRecordContextParams<Customer>) => {
+  const record = useRecordContext<Customer>(props);
```
Comment (slax57) [inline src/ressources/customers/ReviewsNbField.tsx]: Don't need to pass props

File: src/ressources/customers/ReviewsNbField.tsx [hunk 13]
```diff
@@ -0,0 +1,31 @@
...
+  const { data, total, isLoading, error } = useGetList("reviews", {
+    pagination: { page: 1, perPage: 5 },
+    filter: { customer_id: record.id },
+  });
```
Comment (slax57) [inline src/ressources/customers/ReviewsNbField.tsx]: Don't need `data`.
If you're only interested in the `total`, you may as well request only `perPage: 1` item to optimize.

File: src/ressources/customers/ReviewsNbField.tsx [hunk 14]
```diff
@@ -0,0 +1,31 @@
...
+
+  if (isLoading) {
+    return <Typography variant="body2">Loading</Typography>;
+  }
+  if (error) {
+    return <Typography variant="body2">Error</Typography>;
+  }
```
Comment (slax57) [inline src/ressources/customers/ReviewsNbField.tsx:22]: In a real app you would probably not want to handle errors this way (you loose much information). But maybe it's not something important for the scope of this exercice.

File: src/ressources/customers/CustomerEdit.tsx (hunk 9 above)
Comment (Cimanel) [reply src/ressources/customers/CustomerEdit.tsx]: yep thanks!

File: src/ressources/customers/CustomerEdit.tsx (hunk 11 above)
Comment (Cimanel) [reply src/ressources/customers/CustomerEdit.tsx:116]: No, I don't. As we said, It is not prioritized

**Pull request merged**